spring-cloud-azure-feature-management/src/main/java/com/microsoft/azure/spring/cloud/feature/manager/FeatureManager.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.feature.manager;
+
+import java.util.LinkedHashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Configuration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.spring.cloud.feature.manager.entities.Feature;
+import com.microsoft.azure.spring.cloud.feature.manager.entities.FeatureFilterEvaluationContext;
+import com.microsoft.azure.spring.cloud.feature.manager.entities.FeatureSet;
+
+/**
+ * Holds information on Feature Management properties and can check if a given feature is
+ * enabled.
+ */
+@Configuration
+@ConfigurationProperties(prefix = "feature-management")
```

Comment:
Why is `@ConfigurationProperties` used? Is this an externalized configuration class?

Reply:
If it is, how about adding `Properties` suffix to the class name?

Reply:
The idea here was that Features are read in from either a file or from app-configuration and set to this `FeatureManager`, just like a user could in there own application. We then use the `ApplicationContext` to get the `@Components` they defined for there filters to check the Feature state.